include/deal.II/matrix_free/fe_evaluation_data.h

```diff
@@ -948,7 +963,11 @@ inline FEEvaluationData<dim, Number, is_face>::FEEvaluationData(
   , face_orientation(0)
   , subface_index(0)
   , cell_type(internal::MatrixFreeFunctions::general)
-{}
+{
+  // TODO: This currently fails for simplex/matrix_free_face_integral_01
+  // if (descriptor != nullptr)
+  // AssertDimension(n_quadrature_points, descriptor->n_q_points);
```

Comment:
@peterrum This would now trigger inside `simplex/matrix_free_face_integral` for `WEDGE, degree=2` because `n_quadrature_points` (derived from `shape_info`) would suggest a quadrature formula with 7 points, whereas `descriptor` gets the index 0 and then reports 9 points. Do you know which one is correct? I am currently running an old version to see if this was correct before.

Reply:
This sounds like the triangle (7) and quad (9) faces are mixed up, i.e., the face_type is not set up properly.

Reply:
This assertion would also fail for `master`. It seems the number of quadrature points reported by `shape_info` is not correct. I can work around it for now, but it seems very brittle to rely on different information in different structures for mixed meshes.

Reply:
So in this case it seems the descriptor is correct, as we do not catch this information: https://github.com/dealii/dealii/blob/51ff255b330284f20e88f1bde8a80d4a263abe53/include/deal.II/matrix_free/shape_info.templates.h#L326-L333

Reply:
Oh yes. I remember. There is only one `n_q_points_face`...